File: typescript/9.ts
let prod = 0;
for (let a = 1; a < 1000; a++) {
  for (let b = a; b < 1000 - a; b++) {
    for (let c = b; c < 1000 - b; c++) {
      if (a ** 2 + b ** 2 === c ** 2 && a + b + c === 1000) {
        prod = a * b * c;
      }
    }
  }
}

console.log(prod);
// 31875000

export {};

File: swift/9.swift
import Foundation

var prod = 0

for a in stride(from: 1, to: 1e3, by: 1) {
  for b in stride(from: a, to: 1e3 - a, by: 1) {
    if 500 + (a * b) / 1000 == (a + b) {
      prod = Int(a * b * (1000 - a - b))
    }
  }
}

print(prod)
// 31_875_000
